vital/types/local_cartesian.h

```diff
@@ -0,0 +1,182 @@
+ *    to endorse or promote products derived from this software without specific
+ *    prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE AUTHORS OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * \file
+ * \brief This file contains the interface to a local geographic offset coordinate system.
+ */
+
+#ifndef KWIVER_VITAL_LOCAL_CARTESIAN_H_
+#define KWIVER_VITAL_LOCAL_CARTESIAN_H_
+
+#include <vital/types/point.h>
+#include <vital/types/geo_point.h>
+
+namespace kwiver {
+namespace vital {
+
+// ----------------------------------------------------------------------------
+/** Local Cartesian Conversion Utility.
+ *
+ * Base on the NGA GeoTrans library
+ * https://earth-info.nga.mil/GandG/update/index.php?action=home
+ *
+ * This class is a cleaned up version of the LocalCartesian and Geocentric classes provided in GeoTrans
+ * This allows a user to define a local cartesian coordinate system with any origin (expressed in WGS84)
+ */
+class VITAL_EXPORT local_cartesian
+{
+public:
+
+  local_cartesian();
+  virtual ~local_cartesian() = default;
+
+  /**
+   * Set local origin parameters as inputs and sets the corresponding state variables.
+   * NOTE : If the origin changes, this method needs to be called again to recompute
+   *        variables needed in the conversion math.
+   *
+   *    geo_point                : Geograpical origin of the cartesian system        (input)
+   *    orientation              : Orientation angle of the local cartesian coordinate system,
+   *                               in radians                                        (input)
+   */
+  void set_origin(geo_point const& origin, double orientation=0);
+
+  /**
+   * \brief Get the WGS84 coordinates of the cartesian coordinates.
+   *
+   * \returns The WGS84 lon/lat/alt of the cartesian coordinates.
+   */
+  void convert_from_cartesian(vector_3d const& cartesian_coordinate, geo_point& location) const;
+
+  /**
+   * \brief Set the cartesian coordinates based on 2 geo_points
+   */
+  void convert_to_cartesian(geo_point const& location, vector_3d& cartesian_coordinate) const;
+
+  ////////////////////////////////////////////////////////////////////////////////////
+  // METHODS BELOW NOTICE                                                           //
```

Comment:
I suggest just making these methods `protected`.  Someone can derived a class and get to them if they really need them, but these are really implementation details.  It would be very easy for someone to call the wrong function.

An alternative is to have a utility class that provides all of these helper functions for advanced users that really know how to used them.  However, for the object that gets passed around to represent a the coordinate system, I think a simpler interface is better.  It should really only have local-to-global and global-to-local methods exposed.

Reply:
Yeah, sounds good, I was having the same debate in my head, will do

Reply:
Heck, I think I can put these methods in the pimpl class...

Reply:
Yeah, I think that would be best.  I almost suggested that too.  The only downside is those methods would not exposed anywhere for advanced users.  However, I think we should revisit refactoring for advanced users when someone actually has a need.  I think it's more important for this class to have a clean and intuitive API.

Reply:
Agreed, its pretty easy to expose the pimpl class as a formal vital utility if needed.
But better to wait until then

#[YANGNI](https://martinfowler.com/bliki/Yagni.html)